### ISSUE TITLE
A bug with `n_fused`

### ISSUE THREAD

**Issue by JiayiFeng** [tunib-ai/parallelformers#41]
When a attn_qkv `Layer` is set with `n_fused>1` and `reversed=False`, the shape of its sliced weight is incorrect.

Seems that the root cause is here:

https://github.com/tunib-ai/parallelformers/blob/436573b05b9d47dba2234d60e874ff7bb57b725b/parallelformers/parallel/slicing.py#L79-L95

For a attn_qkv weight, the arg `dim` is 0. So when the `reversed=False` and `n_fused>1`, the tensor is chunked on the dim 0 and then concatenated on the dim 1. Which make its shape incorrect.

**Comment by hyunwoongko** [comment]
which model did you use?

**Comment by JiayiFeng** [comment]
I used a modified GPT-NeoX model, which is not officially supported. So I written a custom policy, and find this issue.

**Comment by JiayiFeng** [comment]
Maybe the 
```python
proj_layer = list( 
         map(lambda x: torch.cat([*x], dim=-1), zip(*proj_layer)) 
     ) 
```
should be:
```python
proj_layer = list( 
         map(lambda x: torch.cat([*x], dim=dim), zip(*proj_layer)) 
     ) 
```
I guess.

**Comment by hyunwoongko** [comment]
okay so could you test it with other models?
